### DISCUSSION TITLE
All devices on my network turned on in the night

### DISCUSSION THREAD

**Discussion by JasGawera** [arendst/Tasmota#22273] [Q&A]
Well, almost all devices - I'll explain what I know...

Last night at 23:45 I was almost asleep when the bedroom lights and fan turned on.
The bedroom has a 4 gang light switch flashed with tasmota. (I think this one is an Athom switch)
It turns out that almost all of these also turned on at the same time.
I used Home Assistant on my phone to switch everything off, and noticed a few that hadn't been switched on.

So this morning I've investigated a bit, here is what I know:

1. Most of my house has sonoff light switches and a few relays, all flashed with tasmota. Mainly 14.1 a few 14.2.
2. All switches are known to Home Assistant
3. Most multi-gang switches have HUE emulation enabled and are advertised to Alexa
4. I have 2 single gang switch that have Belkin Wemo emulation enabled and are advertised to Alexa 
5. I have some switches that have no emulation enabled and thus are not advertised to Alexa
6. All of the hue emulated switches turned on at the same time
7. One of the Wemo switches turned on
8. The other Wemo switch didn't turn on
9. None of the "no emulation" switches turned on

Has anyone seen similar issues?
Any idea how to track down the cause?
I'm not sure how to tell what told the switches to turn on. I don't think it was HA as in point 9 I have switches that didn't come on.
I was thinking only the HUE emulated switches came on, but then I found one of the Wemo ones did too.
Then I was thinking Alexa is the common factor, but found the other wemo one that didn't switch on.

**Comment by sfromis** [comment]
A typical cause of such issues is if you have done the bad thing of having commands to Tasmota published with MQTT retain. Such commands will be saved by the broker, and reissued in cases like a brief network outage, where Tasmota is able to connect again. Note that even if you stopped using retain, the commands are still saved, until/unless you clean them up.
https://tasmota.github.io/docs/FAQ/#my-device-randomly-switches-on-and-off-do-i-have-ghosts-in-my-house
